### ISSUE TITLE
Add support for transport-level encryption to allow Azure usage.

### ISSUE THREAD

**Issue by mp911de** [r2dbc/r2dbc-mssql#5]
Server configurations such as Azure require full transport-level support (e.g. Encryption = required). We should add support for full TLS support.

**Comment by mp911de** [comment]
TLS usage in TDS requires wrapping of SSL packets with a TDS header (prelogin) for the SSL handshake. Once the handshake is finished, SSL data is written directly to the transport.

For Encrypt Off (login-only encryption) we disable SSL. For Encrypt On/Required, we need to reorder ChannelHandlers to be the TdsSslHandler the first one to handle all SSL traffic.

**Comment by Ih8Cocoa** [comment]
When I try to connect to Azure SQL Server using R2DBC, it throws this exception
```
io.netty.handler.codec.DecoderException: io.netty.handler.ssl.NotSslRecordException: not an SSL/TLS record: afcbfe5c0e000000
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:472) ~[netty-codec-4.1.36.Final.jar:4.1.36.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:278) ~[netty-codec-4.1.36.Final.jar:4.1.36.Final]
	at io.r2dbc.mssql.client.ssl.TdsSslHandler.channelRead(TdsSslHandler.java:359) ~[r2dbc-mssql-1.0.0.M7.jar:1.0.0.M7]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:374) ~[netty-transport-4.1.36.Final.jar:4.1.36.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:360) ~[netty-transport-4.1.36.Final.jar:4.1.36.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:352) ~[netty-transport-4.1.36.Final.jar:4.1.36.Final]
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1408) ~[netty-transport-4.1.36.Final.jar:4.1.36.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:374) ~[netty-transport-4.1.36.Final.jar:4.1.36.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:360) ~[netty-transport-4.1.36.Final.jar:4.1.36.Final]
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:930) ~[netty-transport-4.1.36.Final.jar:4.1.36.Final]
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:163) ~[netty-transport-4.1.36.Final.jar:4.1.36.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:682) ~[netty-transport-4.1.36.Final.jar:4.1.36.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:617) ~[netty-transport-4.1.36.Final.jar:4.1.36.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:534) ~[netty-transport-4.1.36.Final.jar:4.1.36.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:496) ~[netty-transport-4.1.36.Final.jar:4.1.36.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:906) ~[netty-common-4.1.36.Final.jar:4.1.36.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.36.Final.jar:4.1.36.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-common-4.1.36.Final.jar:4.1.36.Final]
	at java.base/java.lang.Thread.run(Thread.java:835) ~[na:na]
	Suppressed: reactor.core.publisher.FluxOnAssembly$OnAssemblyException: 
Assembly trace from producer [reactor.core.publisher.FluxConcatMap] :
	reactor.core.publisher.Flux.concatMap(Flux.java:3446)
	io.r2dbc.mssql.client.ReactorNettyClient.<init>(ReactorNettyClient.java:171)
Error has been observed at the following site(s):
	|_ Flux.concatMap ⇢ at io.r2dbc.mssql.client.ReactorNettyClient.<init>(ReactorNettyClient.java:171)
	|_  Flux.doOnNext ⇢ at io.r2dbc.mssql.client.ReactorNettyClient.<init>(ReactorNettyClient.java:186)
```
And the client doesn't get a response
I did use `MssqlConnectionConfiguration.Builder.enableSsl()` on the database configuration method.

**Comment by mp911de** [comment]
Please file a new issue @Ih8Cocoa.
